elastic/eui
# [EuiSearchBar] Allow disabling selection auto sort in `field_value_selection` filters

## Summary

closes https://github.com/elastic/eui/issues/7955

This PR adds a config option to disable auto sort behavior for field value selection filters for EuiSearchBar. In small lists it can be quite annoying to have options jumping around, so it's useful to have a way to disable this. The default behavior is unchanged (selected items will sort to the top of the list). If the optional prop is set to `false`, items will remain in their original order when selected. I modified the cypress test to include a test case for this. This can also be manually tested from the docs site by adding `autoSortOptions: false` to one of the field value toggles.

No options selected
![image](https://github.com/user-attachments/assets/3fc88405-6162-4434-bde2-c2bc9b87c029)

Default behavior
![image](https://github.com/user-attachments/assets/294b7de2-bb9a-49c1-854e-dbaeafeac0fb)

Behavior when `autoSortOptions` is set to false
![image](https://github.com/user-attachments/assets/f90f235c-6cc1-4217-82cd-d62b11fc5d01)


I tried to fill out the checklist, but there were a few items I wasn't sure about so I left them blank.

## QA

- Go to https://eui.elastic.co/pr_7958/#/forms/search-bar#search-bar-filters
- Click `Tag (custom filter)` and then click any option towards the bottom of the list, e.g. `legal`
- [x] Confirm the selected option does not get resorted to the top of the list
- https://eui.elastic.co/pr_7958/storybook/?path=/story/forms-euisearchbar-euisearchbarfilters--playground

### General checklist

- Browser QA
    ~- [ ] Checked in both **light and dark** modes~
    ~- [ ] Checked in **mobile**~
    ~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
- Docs site QA
    - [x] Updated docs/storybook with example of new prop
    ~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
- Code quality checklist
    ~- [ ] Updated **[visual regression tests](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/visual-regression-testing.md)**~
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A
